extensions/extensions-helm/aissemble-infrastructure-chart/templates/namespaces.yaml

```diff
@@ -0,0 +1,31 @@
+---
+{{- if index .Values "argo-cd" "enabled" }}
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: {{ index .Values "argo-cd" "namespaceOverride" }}
+  annotations:
+    "helm.sh/hook": pre-install
+    "helm.sh/hook-weight": "-10"
+---
+{{- end }}
+{{- if .Values.jenkins.enabled }}
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: {{ .Values.jenkins.namespaceOverride }}
+  annotations:
+    "helm.sh/hook": pre-install
+    "helm.sh/hook-weight": "-10"
+---
+{{- end }}
+{{- if index .Values "ingress-nginx" "enabled" }}
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: {{ index .Values "ingress-nginx" "namespaceOverride" }}
```

Comment:
S: I know we're setting the override values in our default, but I think best practice here would still be to default to `Release.Namespace`. This would allow a user to unset these overrides to control the namespace at deployment time. Obviously, if the subcharts don't provide support for `Release.Namespace` it's not useful but most chart developers are trending that direction as a best practice.

Reply:
Yes they all provide support for `Release.Namespace` as their default. Thought behind overriding the namespaces here was so you can deploy the helm chart in one command, but not end up with all your charts in the same namespace. I'll move the guidance to setting the namespaces as part of the helm install command in the docs, so we can preserve the `Release.Namespace` functionality.

Reply:
Due to issues in helm with creating a namespace as part of a helm chart that's the same as the `Release.Namespace` this default could not be included. Instead the namespace is only created when the override is present and differs from the `Release.Namespace`.